src/Model/Payload.php

```diff
@@ -113,4 +110,37 @@ public function getCustomData(): array
     {
         return $this->customData;
     }
+
+    /**
+     * Guard the custom data upon instantiation.
+     *
+     * @param array $data
+     */
+    private function guardCustomData(array $data)
+    {
+        foreach ($data as $key => $value) {
+            $this->validateCustomDataValue($value);
+            $this->validateCustomDataKey($key);
+        }
+    }
+
+    private function validateCustomDataValue($value)
```

Comment:
Please add the correct doc comment for each methods.

Doc comment should contain:

* Comment of method
* Input arguments with types.
* Return declaration
* Throws declaration